### PR TITLE
HARMONY-490: Update documentation for service developers with changes since using Argo to implement workflows.

### PR DESCRIPTION
**Pull request by chris-durbin**

### CODE REVIEW

File: docs/adapting-new-services.md [hunk 1]
```diff
@@ -9,23 +9,23 @@ In order to connect a new service to Harmony:
...
-and Harmony callbacks.  Full details as well as an example can be found in the project's README and code.
+Harmony provides a Python library, [harmony-service-lib-py](https://github.com/nasa/harmony-service-lib-py), to ease the process of adapting Harmony messages to subsetter code. It provides helpers for message parsing, command line interactions, data staging, reading and writing STAC catalogs, and Harmony callbacks. Full details as well as an example can be found in the project's README and code.
 
 ### Docker Container Images
 
-The service and all necessary code and dependencies to allow it to run can be packaged in a Docker container image.  Docker images can be staged anywhere Harmony can reach them, e.g. Dockerhub or AWS ECR.  Harmony will run the Docker image, passing the following command-line parameters:
+The service and all necessary code and dependencies to allow it to run can be packaged in a Docker container image. Docker images can be staged anywhere Harmony can reach them, e.g. Dockerhub or AWS ECR. Harmony will run the Docker image, passing the following command-line parameters:
```
Comment (indiejames) [inline docs/adapting-new-services.md:24]: Change 'can be' to 'must be'

File: docs/adapting-new-services.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ Synchronous requests are ones where a user has made a call to Harmony and the co
 
 #### For Docker services
 
-Once complete, a service must send an HTTP POST request to the URL provided in the `callback` field of the Harmony input.  Failing to do so will cause user requests to hang until a timeout that is likely long in order to accommodate large, synchronous operations.  Please be mindful of this and provide ample error handling.
+Once complete, a service must send an HTTP POST request to the URL provided in the `callback` field of the Harmony input.  Failing to do so will cause user requests to hang until a timeout that is likely long in order to accommodate large, synchronous operations. Please be mindful of this and provide ample error handling.
```
Comment (indiejames) [inline docs/adapting-new-services.md:61]: This isn't true anymore, is it? The result handler sends the responses, right?

File: docs/adapting-new-services.md [hunk 3]
```diff
@@ -261,12 +346,40 @@ original Harmony request.
...
+4. Proactively protect (non-Docker) service endpoints from high request volume or computational requirements by using autoscaling with maximum thresholds, queueing, and other methods to avoid outages or non-responsiveness.
+5. Use the latest available data-operation schema. As harmony development continues the schema will evolve to support new features, which will require services be compatible with the new schema in order to take advantage of those features. In addition code to support older versions of the schema can be retired once all services have updated to later versions of the schema.
+6. Name files according to the [established conventions](https://wiki.earthdata.nasa.gov/pages/viewpage.action?spaceKey=HARMONY&title=Output+File+Naming+Convention).
+Using the Harmony-provided Python library makes this automatic for cases where the file corresponds to a single granule.  Files subset to a single variable should be
+suffixed with underscore followed by the variable name.  Files that have been regridded should be suffixed with `_regridded`.  Files that have been subsetted should
```
Comment (indiejames) [inline docs/adapting-new-services.md:375]: Maybe provide examples?

**Review by indiejames — CHANGES_REQUESTED** [review]
Looks good. Some small nits, but one real question about callbacks.

**Review by indiejames — APPROVED** [review]

File: docs/adapting-new-services.md (hunk 2 above)
Comment (chris-durbin) [reply docs/adapting-new-services.md]: Updated the callback section to clarify what is required now that we're using Argo for docker services.

File: docs/adapting-new-services.md (hunk 1 above)
Comment (chris-durbin) [reply docs/adapting-new-services.md:24]: This section talks about either using docker services or HTTP services so I think can is OK in this context.

File: docs/adapting-new-services.md [hunk 4]
```diff
@@ -9,44 +9,44 @@ In order to connect a new service to Harmony:
...
+The service and all necessary code and dependencies to allow it to run can be packaged in a Docker container image. Docker images can be staged anywhere Harmony can reach them, e.g. Dockerhub or AWS ECR. Harmony will run the Docker image, passing the following command-line parameters:
 
 `--harmony-action <action> --harmony-input <input> --harmony-sources <sources-file>`
 
-`<action>` is the action Harmony wants the service to perform.  Currently, Harmony only uses `invoke`, which requests that the service be run and exit.  The service library Harmony provides also supports a `start` action with parameter `--harmony-queue-url <url>`, which requests that the service be started as a long running service that reads requests from an SQS queue.  This is likely to be deprecated.
+`<action>` is the action Harmony wants the service to perform. Currently, Harmony only uses `invoke`, which requests that the service be run and exit. The service library Harmony provides also supports a `start` action with parameter `--harmony-queue-url <url>`, which requests that the service be started as a long running service that reads requests from an SQS queue. This is likely to be deprecated.
```
Comment (bilts) [inline docs/adapting-new-services.md:28]: Could you please do me a favor and move yourself and these single spaces somewhere that is no longer on my lawn?

**Review by bilts — APPROVED** [review]

**Pull request merged**